client/blocks/signup-form/index.jsx

```diff
@@ -171,11 +141,45 @@ class SignupForm extends Component {
 		const initialState = this.formStateController.getInitialState();
 		const stateWithFilledUsername = this.autoFillUsername( initialState );
 
-		this.maybeRedirectToSocialConnect( this.props );
+		this.state = {
+			submitting: false,
+			isFieldDirty: {
+				email: false,
+				username: false,
+				password: false,
+				firstName: false,
+				lastName: false,
+			},
+			form: stateWithFilledUsername,
```

Comment:
`form: stateWithFilledUsername` used to only be set after calling `maybeRedirectToSocialConnect()`, but I can see no good reason for that. So I think simplifying things and just setting the `form` state in the constructor is the right thing to do.